### PR TITLE
build tutorials

### PR DESCRIPTION
**Pull request by jpchen**
http://pyro.ai/numpyro

TODOs:
 - [x] Move tutorials to source (remove from tutorial/).
 - [x] Remove Minipyro from examples.

Other TODOs (not necessarily this PR):
 - [x] Recognizing bullets / numbering to render on separate lines (see tutorial outline for bayesian regression).
 - [x] Math equations not rendered (https://github.com/pyro-ppl/numpyro/pull/197#issuecomment-500566190)

### CODE REVIEW

**Comment by martinjankowiak** [comment]
can you do auto-rendered code examples from python scripts like [this](http://pyro.ai/examples/hmm.html) one too?

File: notebooks/source/index.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Welcome to NumPyro Examples and Tutorials!
+==========================================
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Examples:
+
+   bayesian_regression
+   logistic_regression
```
Comment (neerajprad) [inline notebooks/source/index.rst]: Can we remove logistic_regression for now? This was mostly used for internal benchmarking and is not likely to be useful as a reference, unless we add more description to it and carefully compare performance across implementations in other frameworks (right now we are just comparing with reported numbers in the paper).

**Comment by neerajprad** [comment]
Many thanks for exposing this! Just some minor comments:
 - The tutorial table of contents doesn't seem to be rendered correctly for `bayesian_regression`.
 - Reg. Martin's comment, can we also render the examples from the `examples` directory?

**Comment by jpchen** [comment]
> The tutorial table of contents doesn't seem to be rendered correctly for bayesian_regression.

how do you mean? it looks fine to me?

> Reg. Martin's comment, can we also render the examples from the examples directory?

yeah im working on that, getting some weird cryptic sphinx errors

**Comment by neerajprad** [comment]
ToC rendered:

Screen Shot 2019-06-09 at 7 06 48 PM


Should be (from notebook):

Screen Shot 2019-06-09 at 7 07 14 PM

**Comment by jpchen** [comment]
not sure if it the css didnt load for you? from the website: 
Screen Shot 2019-06-09 at 7 14 37 PM

**Comment by neerajprad** [comment]
The sidebar is fine - this is the outline from the tutorial introduction (first para). I think it's just some issue with the lists / bullets not being correctly recognized.

**Comment by fehiepsi** [comment]
Thanks @jpchen ! The page looks beautiful!

**Comment by fehiepsi** [comment]
Can we unify `notebooks/source/*.ipynb` and `notebooks/*.ipynb` so we don't have to worry about the duplicated one when we change the content of a notebook?

**Comment by neerajprad** [comment]
> yeah im working on that, getting some weird cryptic sphinx errors

I see the examples being rendered now. Can we also remove the `minipyro.py` example - I think it is misleadingly named, and it might be better to have a separate example with the pyro.generic interface instead?

**Comment by jpchen** [comment]
> Can we unify notebooks/source/*.ipynb and notebooks/*.ipynb so we don't have to worry about the duplicated one when we change the content of a notebook?

this PR moves all the notebooks to `source/`

**Comment by fehiepsi** [comment]
> this PR moves all the notebooks to `source/`

In that case, I think we need to delete notebooks in [notebooks](https://github.com/pyro-ppl/numpyro/tree/3112e235a09e5a6f75368c308bd0f9be4dc1d85d/notebooks) folder too. Could you also update links in README to point to the new address? :)

**Comment by fehiepsi** [comment]
It seems that http://pyro.ai/numpyro/time_series_forecasting.html#Model is not rendered with `nbsphinx-math`. That might be a mistake from me, so I'll try to fix it. :)

@jpchen @martinjankowiak @neerajprad what do you think about add pictures at the top of each example code? I think that the plots of `bnn` and `gp` are pretty nice to be included. :)

**Comment by jpchen** [comment]
> Could you also update links in README to point to the new address?

done

> It seems that... is not rendered with nbsphinx-math

done

> what do you think about add pictures at the top of each example code? I think that the plots of bnn and gp are pretty nice to be included.

i'll leave that to you guys, as long as you update the respective files, i can build and deploy them

**Comment by fehiepsi** [comment]
Thanks @jpchen ! About the pictures, that is just a related question for discussion, and it should be addressed separately. :)

**Comment by neerajprad** [comment]
@jpchen - I just added a few issues to the PR description. This should be good to merge once Minipyro is removed. We can handle the remaining issues in separate PRs and Du and I can take a look at that, but based on your experience with Pyro, do you have any idea why bullets/numbering and math equations are not being recognized?

**Comment by neerajprad** [comment]
> i'll leave that to you guys, as long as you update the respective files, i can build and deploy them

I think @fehiepsi is talking about example scripts that generate separate matplotlib plots as a pdf for example. Is it straightforward to render that at the bottom of the examples (not tutorials, but python scripts)?

**Comment by jpchen** [comment]
i think i addressed all those bullet points (i havent updated the website, i was going to do it after merge but can do it now so you can see)

**Comment by neerajprad** [comment]
I think we'll need #202 merged for the build to pass.

**Pull request merged**